org.eclipse.scout.sdk.core.s/src/main/java/org/eclipse/scout/sdk/core/s/permission/PermissionGenerator.java

```diff
@@ -10,33 +10,82 @@
 package org.eclipse.scout.sdk.core.s.permission;
 
 import static org.eclipse.scout.sdk.core.util.Ensure.newFail;
+import static org.eclipse.scout.sdk.core.util.Strings.isBlank;
 
-import org.eclipse.scout.sdk.core.java.generator.IJavaElementGenerator;
+import java.util.Optional;
+
+import org.eclipse.scout.sdk.core.java.JavaTypes;
 import org.eclipse.scout.sdk.core.java.generator.field.FieldGenerator;
 import org.eclipse.scout.sdk.core.java.generator.method.IMethodGenerator;
 import org.eclipse.scout.sdk.core.java.generator.method.MethodGenerator;
 import org.eclipse.scout.sdk.core.java.generator.type.PrimaryTypeGenerator;
 import org.eclipse.scout.sdk.core.s.java.apidef.IScoutApi;
+import org.eclipse.scout.sdk.core.s.java.generator.method.ScoutFieldGenerator;
+import org.eclipse.scout.sdk.core.util.Strings;
 
-/**
- * <h3>{@link PermissionGenerator}</h3>
- *
- * @since 5.2.0
- */
-public class PermissionGenerator<TYPE extends PermissionGenerator<TYPE>> extends PrimaryTypeGenerator<TYPE> {
+public class PermissionGenerator extends PrimaryTypeGenerator<PermissionGenerator> {
+
+  public static final String ID_FIELD_NAME = "ID";
+
+  private String m_idRef;
+
+  public static PermissionGenerator create() {
+    return new PermissionGenerator();
+  }
 
   @Override
   protected void setup() {
+    withPreProcessor((e, c) -> {
+      withoutField(field -> field.elementName(c).map(ID_FIELD_NAME::equals).orElse(false));
+      idRef()
+          .filter(ID_FIELD_NAME::equals)
+          .ifPresent(idRef -> withIdRef(null));
```

Comment:
why is this necessary?

Reply:
Consider the following: A generator is reused and the first run created an `ID`-constant but for the second run an external reference is passed. In this case we need to remove the `ID`-field before the second run.